src/portal/azext_portal/generated/custom.py

```diff
@@ -0,0 +1,92 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+
+import json
+from knack.util import CLIError
+
+
+def portal_dashboard_list(cmd, client,
+                          resource_group_name=None):
+    if resource_group_name is not None:
```

Comment:
Please use `if resource_group_name` so we can use the `-g ""` workaround to disable the passing of default resource group.

Reply:
I'm not clear what you mean. Here `-g` is an optional parameter, if it is not None will "list by this resource group", otherwise will "list by the subscription" which configured in your az account. I don't see there is a place that should `disable the passing of default resource group`.
Pls correct me if I misunderstood.

Reply:
When users run `az portal dashboard list`, they expect to list by subscription. But if they have run  `az configure -d group=myDefaultResourceGroup`, then the previous command will append `-g myDefaultResourceGroup` by CLI. If users still want to list by subscription in this case, they can use the workaround `az portal dashboard list -g ""` or `az portal dashboard list -g=`. So here we allow users to pass an empty resource group name to avoid appending the default resource group. 

We may want to have an explicit switch to temporarily turn off the default values, but before that is implemented, let's keep the workaround working for list command.

Reply:
@RodgeFu can we make this change in codegen for the list function that could either list by resource group or list by subscription? Use `if resource_group_name` instead of `if resource_group_name is not None`.

Reply:
> When users run `az portal dashboard list`, they expect to list by subscription. But if they have run `az configure -d group=myDefaultResourceGroup`, then the previous command will append `-g myDefaultResourceGroup` by CLI. If users still want to list by subscription in this case, they can use the workaround `az portal dashboard list -g ""` or `az portal dashboard list -g=`. So here we allow users to pass an empty resource group name to avoid appending the default resource group. The original `if resource_group_name is not None` will still list by resource group with the empty resource group, so we'd better use `if resource_group_name`.
> 
> We may want to have an explicit switch to temporarily turn off the default values, but before that is implemented, let's keep the workaround working for list command.

Got it, I didn't know the default rg setting.